File: seongwon/Programmers/Level_0/Lv0_안전지대.java
class Solution {
    public int solution(int[][] board) {
        int answer = 0;
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board[i].length; j++) {
              
                if (board[i][j] == 1) {  // 1인지역 찾음
                  
                    for (int a = i - 1; a <= i + 1; a++) {  //현재위치에서 {-1,-1} 부터 {+1,+1} 까지 위험지역
                        for (int b = j - 1; b <= j + 1; b++) {
                          
                            if (a >= 0 && a < board.length && b >= 0 && b < board[i].length) {  // 배열 벗어나면 안댕
                                if(board[a][b] != 1)  //지뢰가 있는곳은 1
                                board[a][b] = 2;   //지뢰 주위 위험지역은 2
                            }
                        }
                    }
                }
            }
        }
        for (int[] row : board) {
            for (int num : row) {
                if (num == 0) {
                    answer++;
                }
            }
        }
        return answer;
    }
}

File: dohyeon/Programmers/Level_0/Lv0_안전지대.py
def solution(board):
    answer = 0
    arr = [[0 for col in range(len(board))] for row in range(len(board))]
    mine_index = []
    for i in range(len(board)) :
        for j in range(len(board[i])) :
            if board[i][j] == 1 :
                mine_index.append([i, j])
    for i in mine_index :
        x, y = i[0], i[1]
        
        try : arr[x][y] = 1
        except : pass
        if x-1 >= 0 and y -1 >= 0 :
            try : arr[x - 1][y - 1] = 1
            except : pass

        if x - 1 >= 0 :
            try : arr[x - 1][y] = 1
            except : pass

        try : arr[x + 1][y + 1] = 1
        except : pass

        try : arr[x + 1][y] = 1
        except : pass

        if y - 1 >= 0 :
            try : arr[x][y - 1] = 1
            except : pass

        try : arr[x][y + 1] = 1
        except : pass

        if x - 1 >= 0 :
            try : arr[x - 1][y + 1] = 1
            except : pass

        if y - 1 >= 0 :
            try : arr[x + 1][y - 1] = 1
            except : pass
        
      
    for i in range(len(arr)) :
        for j in range(len(arr[i])) :
            print(arr[i][j], end = ' ')
        print()
    
    for i in arr :
        for j in i :
            if j == 0 :
                answer += 1
    
        
    return answer

print(solution([[0, 0, 1], [0, 0, 0], [0, 0, 0]]))